### PR TITLE
Add Annotation, etc. abstract base classes and @abstract decorator

### LINKED ISSUE #2769: Annotations should get an abstract base class

### PR DESCRIPTION
**Pull request by mattpap**
`@abstract` is a phony decorator for now, i.e. it's there to inform you, but it won't automate anything. In future this can be extended to check if abstract base classes aren't initialized.

fixes #2769

### CODE REVIEW

**Comment by canavandl** [comment]
LGTM. ping @mattpap

File: bokeh/models/widgets/markups.py [hunk 1]
```diff
@@ -3,10 +3,15 @@
 """
 from __future__ import absolute_import
 
+from ...properties import abstract
 from ...properties import Int, String
 from ..widget import Widget
 
-class Paragraph(Widget):
+@abstract
+class Markup(Widget):
```
Comment (birdsarah) [inline bokeh/models/widgets/markups.py]: Is this just clean up along the way?

Comment (mattpap) [inline bokeh/models/widgets/markups.py]: Yes (see two lines below), to match with implementation in Scala.

**Comment by bryevdv** [comment]
LGTM

**Comment by mattpap** [comment]
@canavandl, one question remains, what are all those rebases about? You should merge `master` (or equivalent) branch into a published development branch.

**Comment by bryevdv** [comment]
Oh I missed that, I tend to look at the diff, and the report, and local-run output. I missed the rebases. Can you one of you make a clean PR from a diff applied to a new branch off master?

**Comment by canavandl** [comment]
I lied in my commit messages - I actually did merge master into the branch instead of rebasing. I think this PR is good to merge.

**Pull request merged**